### PR TITLE
Image validatition will not consider image in the yml file as error

### PR DESCRIPTION
**Pull request by hod-alpert**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

### CODE REVIEW

File: demisto_sdk/commands/common/hook_validations/image.py [hunk 1]
```diff
@@ -83,10 +83,6 @@ def is_existing_image(self):
 
         if not re.match(INTEGRATION_REGEX, self.file_path, re.IGNORECASE):
             package_path = os.path.dirname(self.file_path)
```
Comment (Itay4) [inline demisto_sdk/commands/common/hook_validations/image.py:85]: need to check the image is exactly in one of: yml or dir

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,6 @@
 # Changelog
 * Added support for IndicatorTypes directory in packs (for `reputation` files, instead of Misc).
+* Image in the yml file will not be considered as error.
```
Comment (Itay4) [inline CHANGELOG.md:3]:
```suggestion
* Improved image validation in YAML files.
```

File: demisto_sdk/commands/common/tests/image_test.py [hunk 3]
```diff
@@ -28,3 +31,26 @@ def test_is_not_default_image():
     image.INTEGRATION_REGXES.append(int_path)
     image_validator = image.ImageValidator(int_path)
     assert image_validator.is_not_default_image() is False
+
+
+def test_is_valid_image_positive(monkeypatch):
```
Comment (Itay4) [inline demisto_sdk/commands/common/tests/image_test.py:36]: add `Given-When-Then` docstring

File: demisto_sdk/commands/common/tests/image_test.py [hunk 4]
```diff
@@ -28,3 +31,26 @@ def test_is_not_default_image():
...
+    structure = mock_structure(file_path=integration_path)
+    monkeypatch.setattr('demisto_sdk.commands.common.hook_validations.image.INTEGRATION_REGXES', [integration_path])
+    # Adding monkey patching this will make image validator behave like this is an integration outside of
+    # pack context and ignore the image that's in the same folder as the file
+    monkeypatch.setattr('demisto_sdk.commands.common.hook_validations.image.INTEGRATION_REGEX', integration_path)
+    validator = IntegrationValidator(structure)
+    assert validator.is_valid_image() is True
+
+
+def test_image_in_both_yml_and_directory(monkeypatch):
```
Comment (Itay4) [inline demisto_sdk/commands/common/tests/image_test.py:49]: add `Given-When-Then` docstring

File: demisto_sdk/commands/common/hook_validations/image.py [hunk 5]
```diff
@@ -80,16 +80,17 @@ def is_existing_image(self):
...
-                            "file, please update the package {}".format(package_path))
-                return False
             image_path = glob.glob(package_path + '/*.png')
             if image_path:
                 is_image_in_package = True
+        if is_image_in_package and is_image_in_yml:
+            print_error(
+                "The file {} has image in both yml and package, remove the 'image' key from the yml file".format(
+                    self.file_path))
+            self._is_valid = False
```
Comment (Itay4) [inline demisto_sdk/commands/common/hook_validations/image.py:92]: do we need to fill this in? not just return false?

File: demisto_sdk/commands/common/hook_validations/image.py [hunk 6]
```diff
@@ -80,16 +80,17 @@ def is_existing_image(self):
...
         if not re.match(INTEGRATION_REGEX, self.file_path, re.IGNORECASE):
             package_path = os.path.dirname(self.file_path)
-            if is_image_in_yml:
-                print_error("You have added an image in the yml "
-                            "file, please update the package {}".format(package_path))
-                return False
             image_path = glob.glob(package_path + '/*.png')
             if image_path:
                 is_image_in_package = True
+        if is_image_in_package and is_image_in_yml:
```
Comment (Itay4) [inline demisto_sdk/commands/common/hook_validations/image.py:88]: i think the condition should be that it is exactly in one, not in both

Comment (hod-alpert) [reply demisto_sdk/commands/common/hook_validations/image.py:88]: I'm returning error if it's in both and if it's in none of the two, if it's in one- no exception will be raised, is that what you mean?

File: demisto_sdk/commands/common/hook_validations/image.py (hunk 5 above)
Comment (hod-alpert) [reply demisto_sdk/commands/common/hook_validations/image.py:92]: How will the user know what went wrong if we don't do this?

Comment (Itay4) [reply demisto_sdk/commands/common/hook_validations/image.py:92]: im asking if by returning `False` does it not set `_is_valid` to `False` ?

Comment (Itay4) [reply demisto_sdk/commands/common/hook_validations/image.py:92]: yea, missed second condition
looks good

File: demisto_sdk/commands/common/hook_validations/image.py (hunk 6 above)
Comment (Itay4) [reply demisto_sdk/commands/common/hook_validations/image.py:88]: yea, missed second condition
looks good

File: demisto_sdk/commands/common/hook_validations/image.py (hunk 5 above)
Comment (hod-alpert) [reply demisto_sdk/commands/common/hook_validations/image.py:92]: It doesn't set _is_valid to False

**Pull request merged**